test/spv-node-sync-test.js

```diff
@@ -0,0 +1,233 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('./util/assert');
+const consensus = require('../lib/protocol/consensus');
+const FullNode = require('../lib/node/fullnode');
+const SPVNode = require('../lib/node/spvnode');
+
+const ports = {
+  full: {
+    p2p: 49331,
+    node: 49332,
+    wallet: 49333
+  },
+  spv: {
+    p2p: 49431,
+    node: 49432,
+    wallet: 49433
+  }
+};
+
+const node = new FullNode({
+  network: 'regtest',
+  workers: true,
+  listen: true,
+  bip37: true,
+  port: ports.full.p2p,
+  httpPort: ports.full.node,
+  maxOutbound: 1,
+  seeds: [],
+  memory: true,
+  plugins: [require('../lib/wallet/plugin')],
+  env: {
+    'BCOIN_WALLET_HTTP_PORT': (ports.full.wallet).toString()
+  }
+});
+
+const spvnode = new SPVNode({
+  network: 'regtest',
+  workers: true,
+  listen: true,
+  port: ports.spv.p2p,
+  httpPort: ports.spv.node,
+  maxOutbound: 1,
+  seeds: [],
+  nodes: [`127.0.0.1:${ports.full.p2p}`],
+  memory: true,
+  plugins: [require('../lib/wallet/plugin')],
+  env: {
+    'BCOIN_WALLET_HTTP_PORT': (ports.spv.wallet).toString()
+  }
+});
+
+const chain = node.chain;
+const miner = node.miner;
+const {wdb} = node.require('walletdb');
+const {wdb: spvwdb} = spvnode.require('walletdb');
+
+let wallet = null;
+let spvwallet = null;
+let spvaddr = null;
+let tip1 = null;
+let tip2 = null;
+
+async function mineBlock(tip) {
+  const job = await miner.createJob(tip);
+  return await job.mineAsync();
+}
+
+async function event(obj, name) {
```

Comment:
I should add this to `test/util/common.js` in https://github.com/bcoin-org/bcoin/pull/748, so it'll be available for other tests.

Reply:
I agree, it is a good test utility function

Reply:
If you think #748 will get merged first, I'll wait and rebase and use `common` as a dependency. Or if this gets merged first, you can move it to `common` in #748 ?

Reply:
Okay, I added it to https://github.com/bcoin-org/bcoin/pull/748